src/models/TranslationKey/addTranslationKeyToProject/index.js

```diff
@@ -0,0 +1,68 @@
+// @flow
+import TranslationKey from '@models/TranslationKey'
+import type { ValidationError } from '@types/ValidationError'
+import validateString from '@helpers/validateString'
+import Project from '@models/Project'
+import { DUPLICATE_ENTRY_ERROR_TYPE } from '@constants/errors'
+
+const validateKey = validateString({
+  type: 'translation key',
+  minLength: 1,
+  maxLength: 55
+})
+
+export type AddTranslationKeyToProjectArgs = {
+  key: string,
+  projectId: string,
+  ownerId: string
+}
+
+type AddTranslationKeyToProjectResponse = {
+  project?: Project,
+  errors: Array<ValidationError>
+}
+
+const addTranslationKeyToProject = async ({
+  key,
+  projectId,
+  ownerId
+}: AddTranslationKeyToProjectArgs): Promise<
+  AddTranslationKeyToProjectResponse
+> => {
+  const errors: Array<ValidationError> = []
+  const keyValidation = validateKey(key)
+
+  if (!keyValidation.isValid) {
+    errors.push({ field: 'key', message: keyValidation.error })
+
+    return { errors }
+  }
+
+  //  First some authorization stuff
+  const project: Project = await Project.query()
+    .findById(projectId)
+    .eager('translationKeys')
+
+  if (!project || (project && !project.hasOwnerId(ownerId))) {
```

Comment:
The `project &&` is redundant since the other side of the OR conditional failed.